src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -141,14 +141,31 @@ internal IEnumerable<FileInfo> GetRestoreFiles()
 
         internal IEnumerable<FileInfo> GetBuildFiles()
         {
-            return GetInventory(_inventoryFiles.Build, GetRestoreFiles, DoBuild);
+            return GetInventory(
+                _inventoryFiles.Build,
+                () =>
+                {
+                    var preInventory = new List<FileInfo>(GetRestoreFiles());
```

Comment:
Why is this needed? The previous logic would just take a before/after snapshot and diff. It didn't take dependencies on the order of events for the calculation. Only the execution of a command caused its predecessor to be executed. I don't like the duplication of knowledge.